lib/logflare/backends/adaptor/postgres_adaptor/repo/migrations/add_log_events.ex

```diff
@@ -1,16 +1,36 @@
 defmodule Logflare.Backends.Adaptor.PostgresAdaptor.Repo.Migrations.AddLogEvents do
+  @moduledoc """
+  Migration to generate a log_events table for a given source_id
+  """
   use Ecto.Migration
+  alias Logflare.Backends.SourceBackend
 
-  def up do
-    create table(:log_events, primary_key: false) do
-      add(:id, :string, primary_key: true)
-      add(:body, :map)
-      add(:event_message, :string)
-      add(:timestamp, :utc_datetime_usec)
+  def generate_migration(%SourceBackend{source_id: source_id}) do
+    name = Module.concat([__MODULE__, "MigrationFor#{source_id}"])
+
+    ast =
+      quote do
+        use Ecto.Migration
+
+        def up do
+          create table("log_events_#{unquote(source_id)}", primary_key: false) do
+            add(:id, :string, primary_key: true)
+            add(:body, :map)
```

Comment:
what is this field? it shouldn't be present by default for table creation, only id, event_message, and timestamp.
unless we are storing the entire log event as a jsonb? if so, then we would be duplicating fields.

Reply:
if schema changing is coming after, then it is fine to just focus on the 3 main fields (and drop body), since schema diffing would require generating migrations on each schema change.

Reply:
this is to store the entire content which is the only way I can imagine doing endpoints properly since most of the queries will require that 🤔 event_message won't be enough to replicate the endpoints behaviours we have for CLI development

Reply:
having top level keys as columns would still be needed for partitioning for performance, as well as indexing, even if we can index jsonb keys. Something to consider, though it definitely is simpler to throw everything into a jsonb column.
since this is the mvp, we can stick with the `body` column approach, though if that is so then we would need to make the postgres table schema that we set as transparent to the user as possible, so that there is less confusion if they compare how data is stored to BigQuery.  we can probably do some sql parsing magic to achieve this.

Reply:
> though it definitely is simpler to throw everything into a jsonb column.

this is the body column at the moment, just a jsonb with everything for now and then we can refine

I do agree that for now we can stay with this since it's so walled off and iterate on it further